python/dgl/distributed/graph_partition_book.py

```diff
@@ -738,14 +777,32 @@ def canonical_etypes(self):
         """
         return [DEFAULT_ETYPE]
 
+    def to_canonical_etype(self, etype):
+        """Convert an edge type to the corresponding canonical edge type.
+
+        Parameters
+        ----------
+        etype : str or (str, str, str)
+            The edge type
+
+        Returns
+        -------
+        (str, str, str)
+            The corresponding canonical edge type
+        """
+        assert etype in (
+            DEFAULT_ETYPE,
+            DEFAULT_ETYPE[1],
+        ), "Base partition book only supports homogeneous graph."
+        return self.canonical_etypes
```

Comment:
This one looks wrong to me.
It returns canonical_etypes, while the etype is not used to get canonical_etype.

Reply:
It's correct as it's `BasicPartitionBook`. canonical etype is always `('_N', '_E', '_N')`